### ISSUE TITLE
Segmentation fault in Alpine

### ISSUE THREAD

**Issue by yoonwaiyan** [anycable/anycable#47]
Hi, we're trying to Dockerize our Rails application which is using anycable to broadcast status updates, unfortunately when I try to start our DelayedJob, it's failed with this error: 

```
job_1  | /usr/local/bundle/gems/anycable-0.5.1/lib/anycable/rpc/rpc_pb.rb:49: [BUG] Segmentation fault at 0x0000000000016d96
job_1  | ruby 2.4.4p296 (2018-03-28 revision 63013) [x86_64-linux-musl]
job_1  |
job_1  | -- Control frame information -----------------------------------------------
job_1  | c:0066 p:---- s:0322 e:000321 CFUNC  :msgclass
job_1  | c:0065 p:0026 s:0318 e:000317 CLASS  /usr/local/bundle/gems/anycable-0.5.1/lib/anycable/rpc/rpc_pb.rb:49
job_1  | c:0064 p:0039 s:0315 E:001880 TOP    /usr/local/bundle/gems/anycable-0.5.1/lib/anycable/rpc/rpc_pb.rb:48 [FINISH]
job_1  | c:0063 p:---- s:0312 e:000311 CFUNC  :require
job_1  | c:0062 p:0012 s:0307 e:000306 BLOCK  /usr/local/bundle/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:293
job_1  | c:0061 p:0057 s:0304 e:000303 METHOD /usr/local/bundle/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:259
job_1  | c:0060 p:0017 s:0298 e:000297 METHOD /usr/local/bundle/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:293
job_1  | c:0059 p:0017 s:0292 e:000291 TOP    /usr/local/bundle/gems/anycable-0.5.1/lib/anycable/rpc_handler.rb:4 [FINISH]
job_1  | c:0058 p:---- s:0289 e:000288 CFUNC  :require
job_1  | c:0057 p:0012 s:0284 e:000283 BLOCK  /usr/local/bundle/gems/activesupport-5.0.2/lib/active_support/dependencies.rb:293
... the log is very long so it's hidden for now
```

Not sure if this helps. We're using alpine Docker image and we have to install `libc6-compat` as mentioned in https://github.com/grpc/grpc/issues/6126 for the same error happened to the grpc ruby library. 

Any help is appreciated, thanks in advance.

**Comment by palkan** [comment]
Hi!

I've seen this before)

The problem was caused by `google-protobuf` gem, which used precompiled extensions not-compatible with alpine.

The solution I'd ended up is here: https://gist.github.com/palkan/a19f7853786d83e0822becfc250cc80e (TL;DR you have to build the `google-protobuf` gem from source)

**Comment by yoonwaiyan** [comment]
It does the trick, thanks!

tl;dr: add protobuf gem:
```
gem 'google-protobuf', github: 'google/protobuf'
```
